app-authz-rest-springboot/README.md

```diff
@@ -66,6 +66,12 @@ Make sure your Keycloak server is running on <http://localhost:8180/>. For that,
    
    ````
 
+You should also deploy some JS policies into the Keycloak Server. For that, perform the following steps:
+
+   ````
+   mvn -f ../authz-js-policies clean install && cp ../authz-js-policies/target/authz-js-policies.jar {KEYCLOAK_HOME}/standalone/deployments
+   ````
+
```

Comment:
Same for
```bash
curl -v -X GET \
  http://localhost:8080/api/resourcea \
  -H "Authorization: Bearer "$access_token
```

command below (would drop the `-v -X GET  ` part) so it prints just `Access Granted`:
```
$ curl http://localhost:8080/api/resourcea   -H "Authorization: Bearer "$access_token
Access Granted
```

Reply:
@pedroigor (Sorry if this another newbie question :(, but for the `app-authz-rest-springboot` QS, how is that `${rpt}` expected to work):

IOW after executing the recommended command:
```
[iankko@localhost tmp]$ curl -X POST \
> http://localhost:8180/auth/realms/spring-boot-quickstart/protocol/openid-connect/token \
> -H "Authorization: Bearer "$access_token \
> --data "grant_type=urn:ietf:params:oauth:grant-type:uma-ticket" \
> --data "audience=app-authz-rest-springboot" \
> --data "permission=Default Resource"
{"upgraded":false,"access_token":"eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJGSjg2R2NGM2pUYk5MT2NvNE52WmtVQ0lVbWZZQ3FvcXRPUWVNZmJoTmxFIn0.eyJleHAiOjE1ODYyNTkyMjgsImlhdCI6MTU4NjI1ODkyOCwianRpIjoiMDk0ZTUwOGUtZTk0OS00ZjhkLWFlYjItN2E1OTIzNzg2MmNkIiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo4MTgwL2F1dGgvcmVhbG1zL3NwcmluZy1ib290LXF1aWNrc3RhcnQiLCJhdWQiOiJhcHAtYXV0aHotcmVzdC1zcHJpbmdib290Iiwic3ViIjoiOTBhNjE1ZjItMDExMi00YzBlLThhMDUtZWQzN2VlN2FjOTU2IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoiYXBwLWF1dGh6LXJlc3Qtc3ByaW5nYm9vdCIsInNlc3Npb25fc3RhdGUiOiI0Zjg1ODRiZi02ZTJiLTQ1YzQtODU3ZC1iYzRhY2I5ODBjMjAiLCJhY3IiOiIxIiwiYWxsb3dlZC1vcmlnaW5zIjpbImh0dHA6Ly9sb2NhbGhvc3Q6ODA4MCJdLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsidXNlciJdfSwiYXV0aG9yaXphdGlvbiI6eyJwZXJtaXNzaW9ucyI6W3sicnNpZCI6ImQyOWIyNjdhLTFlYWEtNDY2OS1hOWI5LTQ1NTUxNmViY2FlYSIsInJzbmFtZSI6IkRlZmF1bHQgUmVzb3VyY2UifV19LCJzY29wZSI6InByb2ZpbGUgZW1haWwiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsInByZWZlcnJlZF91c2VybmFtZSI6ImFsaWNlIn0.ZRhtOri5qD-cINO4duxnoD5OrZ3OxrGtn4CcCltnLg-a9GDVqhBRSkNIc_Z0BQ6SQSo1OCqMNYf27jW3tzsXwyUHuAOyjyRnp9N8kAg3Ea16zlZ9yVo4l4eQV-Xi4DrLGLLoWjCFcnoPT29WKAymA12cDROGwp9-lHaNbkOX-f0","expires_in":300,"refresh_expires_in":1800,"refresh_token":"eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJjZDkzNDZmMi01NWY2LTRkNzUtOThiZi0zYTg3OWM3OTRkNWEifQ.eyJleHAiOjE1ODYyNjA3MjgsImlhdCI6MTU4NjI1ODkyOCwianRpIjoiNmI4YWJjNDYtMTdmOC00MDcyLTllMmEtMzQ1NGU5NjQyMWY2IiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo4MTgwL2F1dGgvcmVhbG1zL3NwcmluZy1ib290LXF1aWNrc3RhcnQiLCJhdWQiOiJodHRwOi8vbG9jYWxob3N0OjgxODAvYXV0aC9yZWFsbXMvc3ByaW5nLWJvb3QtcXVpY2tzdGFydCIsInN1YiI6IjkwYTYxNWYyLTAxMTItNGMwZS04YTA1LWVkMzdlZTdhYzk1NiIsInR5cCI6IlJlZnJlc2giLCJhenAiOiJhcHAtYXV0aHotcmVzdC1zcHJpbmdib290Iiwic2Vzc2lvbl9zdGF0ZSI6IjRmODU4NGJmLTZlMmItNDVjNC04NTdkLWJjNGFjYjk4MGMyMCIsImF1dGhvcml6YXRpb24iOnsicGVybWlzc2lvbnMiOlt7InJzaWQiOiJkMjliMjY3YS0xZWFhLTQ2NjktYTliOS00NTU1MTZlYmNhZWEiLCJyc25hbWUiOiJEZWZhdWx0IFJlc291cmNlIn1dfSwic2NvcGUiOiJwcm9maWxlIGVtYWlsIn0.gwCtbiir_P-cwvAMqxoTf2Cw56mXpjz4xIdJz4hHvog","token_type":"Bearer","not-before-policy":0}
```
what is the `${rpt}` value, I was supposed to get? Is it the access_token again?

Reply:
@iankko RPT is basically an access token with some additional authorization info. You obtain RPT when you use `grant type=urn:ietf:params:oauth:grant-type:uma-ticket`. In README you can see it in this snippet:
```
{
    "access_token": "${rpt}",
}
```
So in the response you sent the rpt is in `"access_token":"....."`

> [iankko@localhost tmp]$ curl -X POST \
> http://localhost:8180/auth/realms/spring-boot-quickstart/protocol/openid-connect/token \
> -H "Authorization: Bearer "$access_token \
> --data "grant_type=urn:ietf:params:oauth:grant-type:uma-ticket" \
> --data "audience=app-authz-rest-springboot" \
> --data "permission=Default Resource"
{"upgraded":false,"access_token":"eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJGSjg2R2NGM2pUYk5MT2NvNE52WmtVQ0lVbWZZQ3FvcXRPUWVNZmJoTmxFIn0.eyJleHAiOjE1ODYyNTkyMjgsImlhdCI6MTU4NjI1ODkyOCwianRpIjoiMDk0ZTUwOGUtZTk0OS00ZjhkLWFlYjItN2E1OTIzNzg2MmNkIiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo4MTgwL2F1dGgvcmVhbG1zL3NwcmluZy1ib290LXF1aWNrc3RhcnQiLCJhdWQiOiJhcHAtYXV0aHotcmVzdC1zcHJpbmdib290Iiwic3ViIjoiOTBhNjE1ZjItMDExMi00YzBlLThhMDUtZWQzN2VlN2FjOTU2IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoiYXBwLWF1dGh6LXJlc3Qtc3ByaW5nYm9vdCIsInNlc3Npb25fc3RhdGUiOiI0Zjg1ODRiZi02ZTJiLTQ1YzQtODU3ZC1iYzRhY2I5ODBjMjAiLCJhY3IiOiIxIiwiYWxsb3dlZC1vcmlnaW5zIjpbImh0dHA6Ly9sb2NhbGhvc3Q6ODA4MCJdLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsidXNlciJdfSwiYXV0aG9yaXphdGlvbiI6eyJwZXJtaXNzaW9ucyI6W3sicnNpZCI6ImQyOWIyNjdhLTFlYWEtNDY2OS1hOWI5LTQ1NTUxNmViY2FlYSIsInJzbmFtZSI6IkRlZmF1bHQgUmVzb3VyY2UifV19LCJzY29wZSI6InByb2ZpbGUgZW1haWwiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsInByZWZlcnJlZF91c2VybmFtZSI6ImFsaWNlIn0.ZRhtOri5qD-cINO4duxnoD5OrZ3OxrGtn4CcCltnLg-a9GDVqhBRSkNIc_Z0BQ6SQSo1OCqMNYf27jW3tzsXwyUHuAOyjyRnp9N8kAg3Ea16zlZ9yVo4l4eQV-Xi4DrLGLLoWjCFcnoPT29WKAymA12cDROGwp9-lHaNbkOX-f0","expires_in":300,"refresh_expires_in":1800,"refresh_token":"eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJjZDkzNDZmMi01NWY2LTRkNzUtOThiZi0zYTg3OWM3OTRkNWEifQ.eyJleHAiOjE1ODYyNjA3MjgsImlhdCI6MTU4NjI1ODkyOCwianRpIjoiNmI4YWJjNDYtMTdmOC00MDcyLTllMmEtMzQ1NGU5NjQyMWY2IiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo4MTgwL2F1dGgvcmVhbG1zL3NwcmluZy1ib290LXF1aWNrc3RhcnQiLCJhdWQiOiJodHRwOi8vbG9jYWxob3N0OjgxODAvYXV0aC9yZWFsbXMvc3ByaW5nLWJvb3QtcXVpY2tzdGFydCIsInN1YiI6IjkwYTYxNWYyLTAxMTItNGMwZS04YTA1LWVkMzdlZTdhYzk1NiIsInR5cCI6IlJlZnJlc2giLCJhenAiOiJhcHAtYXV0aHotcmVzdC1zcHJpbmdib290Iiwic2Vzc2lvbl9zdGF0ZSI6IjRmODU4NGJmLTZlMmItNDVjNC04NTdkLWJjNGFjYjk4MGMyMCIsImF1dGhvcml6YXRpb24iOnsicGVybWlzc2lvbnMiOlt7InJzaWQiOiJkMjliMjY3YS0xZWFhLTQ2NjktYTliOS00NTU1MTZlYmNhZWEiLCJyc25hbWUiOiJEZWZhdWx0IFJlc291cmNlIn1dfSwic2NvcGUiOiJwcm9maWxlIGVtYWlsIn0.gwCtbiir_P-cwvAMqxoTf2Cw56mXpjz4xIdJz4hHvog","token_type":"Bearer","not-before-policy":0}

Reply:
@mhajas Ok, got it thanks. You are correct, it's a bit longer after the reply:

```
 echo $access_token | wc -m
896
$ echo $reply_token | wc -m
1088
```

Why not to change the command to:
```
 export rpt_token=$(curl -X POST \
 http://localhost:8180/auth/realms/spring-boot-quickstart/protocol/openid-connect/token \
 -H "Authorization: Bearer "$access_token \
 --data "grant_type=urn:ietf:params:oauth:grant-type:uma-ticket" \
 --data "audience=app-authz-rest-springboot" \
  --data "permission=Default Resource" | jq --raw-output '.access_token' \
 )
```
so it can be reused later without copying it manually?

@pedroigor WDYT -^ ?

Reply:
Done